lib/kernel/read_image.cl

```diff
@@ -373,46 +388,186 @@ read_pixel_linear_3d_float (float4 abc, float4 one_m, int4 ijk0, int4 ijk1,
+                                                order, data));
+              base_index -= ijk0.x;
+            }
+
+          // + a * b * (1 – c) * Ti1j1k0
+          //      + abc.x * abc.y * one_m.z
+          //            * pocl_read_pixel_fast_f ((int4) (ijk1.x, ijk1.y,
+          //            ijk0.z, 0),
+          //                                      width, height, depth,
+          //                                      row_pitch,
+          //                                      slice_pitch, channel_type,
+          //                                      order, data)
+          if (ijk1_x_OK)
+            {
+              base_index += ijk1.x;
+              sum += (abc.x * abc.y * one_m.z
+                      * pocl_read_pixel_fast_f (base_index, channel_type,
+                                                order, data));
+              base_index -= ijk1.x;
+            }
+
+          base_index -= (ijk1.y * row_pitch);
+        }
+
+      base_index -= (ijk0.z * slice_pitch);
+    }
+
+  if (ijk1.z >= 0 && ijk1.z < depth)
+    {
+      base_index += (ijk1.z * slice_pitch);
+
+      if (ijk0_y_OK)
+        {
+          base_index += (ijk0.y * row_pitch);
+
+          // + (1 – a) * (1 – b) * c * Ti0j0k1
+          //      + one_m.x * one_m.y * abc.z
+          //            * pocl_read_pixel_fast_f ((int4) (ijk0.x, ijk0.y,
+          //            ijk1.z, 0),
+          //                                      width, height, depth,
+          //                                      row_pitch,
+          //                                      slice_pitch, channel_type,
+          //                                      order, data)
+          if (ijk0_x_OK)
+            {
+              base_index += ijk0.x;
+              sum += (one_m.x * one_m.y * abc.z
+                      * pocl_read_pixel_fast_f (base_index, channel_type,
+                                                order, data));
+              base_index -= ijk0.x;
+            }
+
+          // + a * (1 – b) * (1 – c) * Ti1j0k0
+          //      + abc.x * one_m.y * abc.z
+          //            * pocl_read_pixel_fast_f ((int4) (ijk1.x, ijk0.y,
+          //            ijk1.z, 0),
+          //                                      width, height, depth,
+          //                                      row_pitch,
+          //                                      slice_pitch, channel_type,
+          //                                      order, data)
+          if (ijk1_x_OK)
+            {
+              base_index += ijk1.x;
+              sum += (abc.x * one_m.y * abc.z
+                      * pocl_read_pixel_fast_f (base_index, channel_type,
+                                                order, data));
+              base_index -= ijk1.x;
+            }
+
+          base_index -= (ijk0.y * row_pitch);
+        }
+
+      if (ijk1_y_OK)
+        {
+          base_index += (ijk1.y * row_pitch);
+
+          // + (1 – a) * b * (1 – c) * Ti0j1k0
```

Comment:
Uhh. The `a * (1 – b) * (1 – c) * Ti1j0k0` lines are from the official formula for linear filtering, but i see i forgot some old commented code there, that should definitely not be there.